src/stdlib/parse_influxdb.rs

```diff
@@ -0,0 +1,404 @@
+                            "host": "A",
+                            "region": "us-west"
+                        },
+                        "timestamp": "2020-05-26T10:26:13.254420Z",
+                        "kind": "absolute",
+                        "gauge": {
+                            "value": 50.5
+                        }
+                    },
+                    {
+                        "name": "cpu_on",
+                        "tags": {
+                            "host": "A",
+                            "region": "us-west"
+                        },
+                        "timestamp": "2020-05-26T10:26:13.254420Z",
+                        "kind": "absolute",
+                        "gauge": {
+                            "value": 1.0
+                        }
+                    },
+                    {
+                        "name": "cpu_sleep",
+                        "tags": {
+                            "host": "A",
+                            "region": "us-west"
+                        },
+                        "timestamp": "2020-05-26T10:26:13.254420Z",
+                        "kind": "absolute",
+                        "gauge": {
+                            "value": 0.0
+                        }
+                    }
+                ]
+            "#}),
+        }]
+    }
+
+    fn compile(
+        &self,
+        _state: &state::TypeState,
+        _ctx: &mut FunctionCompileContext,
+        arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+
+        Ok(ParseInfluxDBFn { value }.as_expr())
+    }
+}
+
+#[derive(Clone, Debug)]
+struct ParseInfluxDBFn {
+    value: Box<dyn Expression>,
+}
+
+impl FunctionExpression for ParseInfluxDBFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let value = self.value.resolve(ctx)?;
+
+        parse_influxdb(value)
+    }
+
+    fn type_def(&self, _: &state::TypeState) -> TypeDef {
+        type_def()
+    }
+}
+
+fn tags_kind() -> Kind {
+    Kind::object(Collection::from_unknown(Kind::bytes()))
+}
+
+fn gauge_kind() -> Kind {
+    Kind::object(BTreeMap::from([("value".into(), Kind::float())]))
+}
+
+fn metric_kind() -> BTreeMap<Field, Kind> {
+    BTreeMap::from([
+        ("name".into(), Kind::bytes()),
+        ("tags".into(), tags_kind()),
+        ("timestamp".into(), Kind::timestamp()),
+        ("kind".into(), Kind::bytes()),
+        ("gauge".into(), gauge_kind()),
+    ])
+}
+
+fn inner_kind() -> Kind {
+    Kind::object(metric_kind())
+}
+
+fn type_def() -> TypeDef {
+    TypeDef::array(Collection::from_unknown(inner_kind())).fallible()
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::btreemap;
+
+    test_function![
+        parse_influxdb => ParseInfluxDB;
+
+        influxdb_valid {
+            args: func_args![ value: format!("cpu,host=A,region=us-west usage_system=64i,usage_user=10u,temperature=50.5,on=true,sleep=false 1590488773254420000") ],
```

Comment:
tried to include all possible value cases in this test, as it will be useful too as an example & benchmark.